sdk/core/azure-core/test/ut/simplified_header.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief makes sure azure/core.hpp can be included.
+ *
+ * @remark This file will catch any issue while trying to use/include the core.hpp header
+ *
+ */
+
+#include "gtest/gtest.h"
+#include <azure/core.hpp>
```

Comment:
As a test to verify that the CI fails, can you please remove one of the headers from core.hpp in this PR and see if and how the CI legs fail, and then revert that change after all the CI legs have finished running?

Reply:
Sure.
If there is a header that does not exist, the compiler will fail saying the header is not found.
If a header is removed (breaking change), the compiler will say that there is an undefined symbol in the test.

I will add one commit for each case to demonstrate

Reply:
Extra non-existing header:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=583929&view=logs&j=fda9a6e2-14f0-5864-d894-5f56622f20fa&t=ca19bb24-dce0-56b6-7e8c-f313cc6f504b

Reply:
Breaking change compiler error when removing a header from core.hpp
https://dev.azure.com/azure-sdk/public/_build/results?buildId=584141&view=logs&jobId=fda9a6e2-14f0-5864-d894-5f56622f20fa&j=fda9a6e2-14f0-5864-d894-5f56622f20fa&t=ca19bb24-dce0-56b6-7e8c-f313cc6f504b

Reply:
I can't see the build pipeline/error for this, it says `Build not found`:
![image](https://user-images.githubusercontent.com/6527137/96798101-ffb49d80-13b4-11eb-9b35-8f3a6a0c9445.png)

Can you retry the failure and let all the legs run fully rather than cancelling it early?

Also, it looks like if this test compiles, it will always run successfully, so the goal here isn't really to test the runtime behavior but just verify that it was able to build/compile and that alone would verify the correctness of the simplified header. Is that correct?

Reply:
I'll approve once I can see what the failure cases look like in CI. Thanks.